sdk/ai/azopenai/example_files_operations_test.go

```diff
@@ -0,0 +1,145 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azopenai_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/ai/azopenai"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/streaming"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+)
+
+func ExampleClient_AddUploadPart() {
+	azureOpenAIKey := os.Getenv("AOAI_API_KEY")
+	azureOpenAIEndpoint := os.Getenv("AOAI_ENDPOINT")
+
+	if azureOpenAIKey == "" || azureOpenAIEndpoint == "" {
+		fmt.Fprintf(os.Stderr, "Skipping example, environment variables missing\n")
+		return
+	}
+
+	keyCredential := azcore.NewKeyCredential(azureOpenAIKey)
+
+	client, err := azopenai.NewClientWithKeyCredential(azureOpenAIEndpoint, keyCredential, nil)
+	if err != nil {
+		fmt.Println("Error creating client:", err)
+		return
+	}
+
+	// Create an upload request
+	createUploadResp, err := client.CreateUpload(context.Background(), azopenai.CreateUploadRequest{
+		Bytes:    to.Ptr(int32(10)),
+		Filename: to.Ptr("test.txt"),
+		MimeType: to.Ptr("text/plain"),
+		Purpose:  to.Ptr(azopenai.CreateUploadRequestPurposeAssistants),
+	}, nil)
+	if err != nil {
+		fmt.Println("Error creating upload:", err)
+		return
+	}
+
+	// Create parts to upload
+	part1 := streaming.NopCloser(strings.NewReader("hello"))
+	part2 := streaming.NopCloser(strings.NewReader("world"))
+
+	// Upload the second part
+	part2Resp, err := client.AddUploadPart(context.Background(), *createUploadResp.ID, part2, nil)
+	if err != nil {
+		fmt.Println("Error uploading part 2:", err)
+		return
+	}
+
+	// Upload the first part
+	part1Resp, err := client.AddUploadPart(context.Background(), *createUploadResp.ID, part1, nil)
+	if err != nil {
+		fmt.Println("Error uploading part 1:", err)
+		return
+	}
+
+	// Complete the upload by specifying the order of parts
+	uploadResp, err := client.CompleteUpload(context.Background(), *createUploadResp.ID, azopenai.CompleteUploadRequest{
+		PartIDs: []string{*part1Resp.ID, *part2Resp.ID},
+	}, nil)
+	if err != nil {
+		fmt.Println("Error completing upload:", err)
+		return
+	}
+
+	// Verify the total size of uploaded parts
+	fmt.Println("Total size of uploaded parts:", *uploadResp.Bytes)
+
+	// Delete the uploaded file
+	_, err = client.DeleteFile(context.Background(), *uploadResp.File.ID, nil)
+	if err != nil {
+		fmt.Println("Error deleting file:", err)
+		return
+	}
+
+	// Output:
```

Comment:
We generally don't make our examples run as tests, so I'd remove the //Output block.